lib/src/resolver.rs

```diff
@@ -28,6 +28,33 @@ impl<M> From<DIDRegistry<M>> for Resolver<M> {
     }
 }
 
+/// Extra context passed to the document builder from the [`Resolver`]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EventContext {
+    /// the timestamp in nanoseconds in which the block from the document was built.
+    pub timestamp: i64,
+}
+
+impl EventContext {
+    pub async fn new<M: Middleware>(
+        meta: &LogMeta,
+        signer: impl Middleware,
+    ) -> Result<Self, ResolverError<M>> {
+        let block = signer
+            .get_block(meta.block_number)
+            .await
+            .map_err(|e| ResolverError::Middleware(e.to_string()))?;
+        let timestamp = block
```

Comment:
This should be consistent by fetching the block time from Ethereum or failing.   Block time can differ from real time.